Code Cleanup

kevinsandow opened the pull request:
I know this is alot... it's mostly linting, sometimes a few type fixes and a couple of files removed from the CRA tool.

kevinsandow commented:
I can also try to split this into multiple PRs, if required.

thmsndk commented:
Can't remember how many spaces I usually have configured.

Would you mind describing what you have removed from the CRA tool? could not see it at a glance.
also looked like you added husky for linting on commit?

kevinsandow commented:
I usually use 2 spaces for indentation, but I don't mind switching to 4, since this is what prettier seemed to be configure to produce, atm.

* changed the title
* removed the SPA files, unless we really want a manifest and a couple of mobile friendly icons looking like react
* removedThe Web Performance Vitals code - I assumed we wanted to get stuff done and performance is rather a secondary concern
* removed remnants of the CRA code, like importing the React logo
* was close to remove Jest as well, but I didn't know whether you guys wanted to test anything

kevinsandow commented:
There also were a log of unused imports I removed, probably from a time where everything was in one file.

thmsndk commented:
I think I usually prefer 2 spaces for indentation as well to not make the code as wide 🤔

kevinsandow commented:
I changed it back to 2 🙂 

That probably means @Telokis prefers 4 over 2.
But this way the changes in the data.json are no longer applicable.

Telokis commented on package.json:
```diff
@@ -25,8 +25,11 @@
-    "lint": "eslint --ext .js,.ts . && echo Linting ok",
-    "lint:fix": "eslint --fix --ext .js,.ts . && echo Linting ok",
+    "format": "npm run format:related .",
+    "format:related": "prettier -u --write",
```
No need to use prettier by itself, it is included in the eslint configuration.

kevinsandow replied:
Ok, I will remove that then...

Telokis commented on .prettierignore:
```diff
@@ -0,0 +1 @@
+build
```
This file is not needed, prettier is not supposed to be used by itself but with eslint.

kevinsandow replied:
Ok, I will remove that then...

Telokis commented on package.json:
```diff
@@ -49,8 +52,10 @@
     "@telokys/eslint-config-react-typescript": "^2.0.5",
     "@types/lz-string": "^1.3.34",
     "eslint": "^8.28.0",
+    "husky": "^8.0.2",
```
I'm not a fan of Husky, I prefer the checks to be on the server side instead.

kevinsandow replied:
Really, so the push would fail, but you already would have a commit with issues.
And this way auto-fixable issues would already be resolved on commit.

kevinsandow replied:
I know you could also configure your IDE to do it on-save and if that is the desired change, I'll remove it, too.

kevinsandow replied:
And gone...

thmsndk commented:
Is this in a state where we can merge it now then? I'm a little hazy where we ended up at :P

Telokis commented:
@thmsndk This is ok for me, yeah
